python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/router_bgp.py

```diff
@@ -36,40 +34,32 @@ def router_bgp(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol) -> dict
                 msg = f"{self.data_model}.p2p_links.[].as or {self.data_model}.p2p_links_profiles.[].as"
                 raise AristaAvdInvalidInputsError(msg)
 
-            neighbor = {
-                "remote_as": p2p_link_data["peer_bgp_as"],
-                "peer": p2p_link_data["peer"],
-                "description": p2p_link_data["peer"],
-                "peer_group": self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
-            }
+            neighbor = EosCliConfigGen.RouterBgp.NeighborsItem(
+                remote_as=p2p_link_data["peer_bgp_as"],
+                peer=p2p_link_data["peer"],
+                description=p2p_link_data["peer"],
+                peer_group=self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
+            )
```

Comment:
you don;t need this here if you hit continue next, put this after the if block for neighbor_interfaces

Reply:
and even afte rthe raise line 58

Reply:
Fixed!